lib/src/list/built_list.dart

```diff
@@ -79,7 +76,11 @@ abstract class BuiltList<E> implements Iterable<E>, BuiltIterable<E> {
     if (identical(other, this)) return true;
     if (other is! BuiltList) return false;
     if (other.length != length) return false;
-    if (other.hashCode != hashCode) return false;
+    if (_hashCode != null &&
+        other._hashCode != null &&
+        _hashCode != other._hashCode) {
+      return false;
+    }
     for (var i = 0; i != length; ++i) {
       if (other[i] != this[i]) return false;
```

Comment:
This is the opposite order of what the platform collections otherwise use. For example. contains will use `this[i] == needle` as a test, not the other way around.
Is there a reason this is not `this[i] == other[i]`?

Reply:
No reason that I can remember :)